solidity/contracts/bridge/WalletProposalValidator.sol

```diff
@@ -608,10 +612,22 @@ contract WalletProposalValidator {
                 "Not a pending redemption request"
             );
 
+            uint32 minAge = REDEMPTION_REQUEST_MIN_AGE;
+            if (redemptionWatchtower != address(0)) {
+                // Check the redemption delay enforced by the watchtower.
+                // slither-disable-next-line calls-loop
+                uint32 delay = IRedemptionWatchtower(redemptionWatchtower)
+                    .getRedemptionDelay(redemptionKey);
+                // If the delay is greater than the usual minimum age, use it.
+                // This way both the min age and the watchtower delay are preserved.
+                if (delay > minAge) {
+                    minAge = delay;
+                }
+            }
+
             require(
                 /* solhint-disable-next-line not-rely-on-time */
-                block.timestamp >
-                    redemptionRequest.requestedAt + REDEMPTION_REQUEST_MIN_AGE,
+                block.timestamp > redemptionRequest.requestedAt + minAge,
```

Comment:
> Are we sure this will not lead to a situation where somebody calls notifyWalletRedemptionTimeout for a redemption that the wallet couldn't process because it had some objections?

Good point! This is not a problem as long as the redemption timeout is greater than the watchtower's level-two delay. In such a case, even if a redemption gets two objections, the wallet can process it before the timeout can be reported.

Reply:
Addressed as part of c6c03a373ac60cd6195b3216da6665e569d1e386